lib/_007/Val.pm

```diff
@@ -205,10 +205,86 @@ class Val::Str does Val {
 ###     say(/"Bond"/.search("J. Bond"));        # --> `True`
 ###
 class Val::Regex does Val {
-    has Val::Str $.contents;
+    # note: a regex should probably keep its lexpad or something to resolve calls&identifiers
+    has $.contents;
 
     method quoted-Str {
-        "/" ~ $.contents.quoted-Str ~ "/"
+        #"/" ~ $.contents.quoted-Str ~ "/"
+        "/...a regex.../"
+    }
+
+    method match(Str $str, Bool $full) {
+        my Int $last-index = 0;
+        sub calibrate($part) {
+            if $last-index == 0 and !$full {
+                $last-index = $str.index($part);
+            }
+        }
+
+        sub parse($fragment) {
+            when $fragment.^name eq "Q::Regex::Str" {
+                my $value = $fragment.contents.value;
+                calibrate($value);
```

Comment:
This sub is only ever called once; maybe it can be inlined here?

But I'm also not sure the logic is right/sufficient. `calibrate` expects to be able to skip to the first matching substring, avoiding all "scanning" along the string in the `search` case. But

    $ bin/007 -e='say(/"bc" "de"/.search("bczzbcde"))'
    False

In this case the first `bc` in the search string is a red herring, and it's the second one we want to find.

I can see how "skip to the first matching constant string" is still a neat time-saver, but I think it breaks down with more elaborate regexes. Consider for example `/ "ab"? "cd" /`. So we probably won't be able to avoid scanning the string character-by-character.

Doesn't stop us from optimizing on top of that basic algorithm, of course! But "first make it work, then make it fast" applies here, I think. :wink: